### PR TITLE
Migration to the new API

### PR DESCRIPTION
**Pull request by maudnals**
* Migrated the code to accommodate header-based source and trigger registration ([Details](https://developer.chrome.com/docs/privacy-sandbox/attribution-reporting-changes-january-2022/))
* Added support for views
* Added support for debug reports

### CODE REVIEW

File: conversion-measurement/adtech/views/ad-view-js.pug [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  meta(name='viewport' content='width=device-width, initial-scale=1.0')
+  title Ad
+body(style={"font-family": "sans-serif", "font-size": "12px", "background": "#f6d7ff"})
+
+div Ad (JS call)
+div#ad(href=href target='_blank')
+  img(width='180' src='/blue-shoes.png' alt='blue shoes')
+  
+script.
+  // \!{}\ is the syntax for this template engine (pug) to access serer-defined variables in a script
```
Comment (apasel422) [inline conversion-measurement/adtech/views/ad-view-js.pug:13]:
```suggestion
  // \!{}\ is the syntax for this template engine (pug) to access server-defined variables in a script
```

Comment (maudnals) [reply conversion-measurement/adtech/views/ad-view-js.pug]: Fixed ty

File: conversion-measurement/adtech/views/ad-click-js.pug [hunk 2]
```diff
@@ -5,27 +5,6 @@ head
...
-  
-  // Display the source event ID in the UI for demo purposes
-  const adIdEl = document.getElementById('adid');
-  adIdEl.innerText = attributionsourceeventid;
-  // Set a click event handler with window.open
-  const adEl = document.getElementById('ad');
-  adEl.addEventListener("click", () => {
-    window.open(href, '_blank', `attributionsourceeventid=${attributionsourceeventid},attributiondestination=${attributiondestination},attributionreportto=${attributionreportto},attributionexpiry=864000000`)
-  });
+div Not supported at the moment
```
Comment (maudnals) [inline conversion-measurement/adtech/views/ad-click-js.pug:8]: (Please ignore this; window.open calls are now supported)

File: conversion-measurement/adtech/server.js [hunk 3]
```diff
@@ -93,39 +165,75 @@ app.get('/conversion', (req, res) => {
...
-  // Adtech orders the browser to schedule-send a report
-  res.redirect(302, url)
+  // Instruct the browser to schedule-send a report
+  res.set(
+    'Attribution-Reporting-Register-Event-Trigger',
+    // ⚠️ Note the enclosing []!
+    JSON.stringify([
+      {
+        trigger_data: `${triggerData}`,
+        debug_key: '123',
```
Comment (maudnals) [inline conversion-measurement/adtech/server.js:188]: Please ignore this. The debug key should be set in the 'Attribution-Reporting-Trigger-Debug-Key' header instead.

**Review by LDbigG — COMMENTED** [review]
gh pr checkout 10

**Review by timscharfenort8 — COMMENTED** [review]
good

**Review by Karinza38 — APPROVED** [review]

**Pull request merged**